### DISCUSSION TITLE
Purpose of the sleep(.2) in the loyalty mod

### DISCUSSION THREAD

**Discussion by Wallboy** [sharkbound/PythonTwitchBotFramework#44] [Q&A] [answered]
Hi!

I've been studying your code and have interest in how the loyalty/point system works in it. I'm curious what the point of the `await sleep(.2)` between adding a balance to each user? 

From my understanding this would result in processing only 300 users per 60 seconds. If your channel had more than 300 chatters, it wouldn't finish processing that batch of users by the time the next loyalty_interval is triggered.

Just curious why you had to include this sleep instead or processing them as fast as possible?

**Comment by sharkbound** [comment, marked as answer]
because disk IO is one of the bigger bottlenecks with any code, i added this in the case that there is a lot of users to process. (to be sure no important async operations are suspended for long)

however thinking about it now, it is a excessive delay, and probably not even needed due to the commit=False.
i will likely remove this
